Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsButtonBase.cs

```diff
@@ -0,0 +1,479 @@
+        /// </summary>
+        protected virtual void ButtonPressed()
+        {
+            OnButtonPressed?.Invoke();
+        }
+
+        /// <summary>
+        /// Invoke the button released event.
+        /// </summary>
+        protected virtual void ButtonUnpressed()
+        {
+            OnButtonUnPressed?.Invoke();
+        }
+
+        /// <summary>
+        /// Handle hand contact with the pressable object.
+        /// </summary>
+        protected virtual void OnHandContactPO(ContactHand hand)
+        {
+            OnHandContact?.Invoke(hand);
+
+            if (hand != null)
+            {
+                if (hand.Handedness == Chirality.Left)
+                {
+                    _leftHandContacting = true;
+                }
+                else if (hand.Handedness == Chirality.Right)
+                {
+                    _rightHandContacting = true;
+                }
+            }
+
+            _contactHandPressing = GetChosenHandInContact();
+        }
+
+        /// <summary>
+        /// Handles actions when a hand exits contact with the pressable object.
+        /// </summary>
+        /// <param name="hand">The hand exiting contact.</param>
+        protected virtual void OnHandContactExitPO(ContactHand hand)
+        {
+            // Invoke event for hand contact exit
+            OnHandContactExit?.Invoke(hand);
+
+            // Update hand contact flags
+            if (hand != null)
+            {
+                if (hand.Handedness == Chirality.Left)
+                {
+                    _leftHandContacting = false;
+                }
+                else if (hand.Handedness == Chirality.Right)
+                {
+                    _rightHandContacting = false;
+                }
+            }
+            else
+            {
+                // Reset hand contact flags if hand is null
+                _leftHandContacting = false;
+                _rightHandContacting = false;
+            }
+
+            // Update overall hand contact flag
+            _contactHandPressing = GetChosenHandInContact();
+        }
+
+        /// <summary>
+        /// Handles actions when a hand exits hovering over the pressable object.
+        /// </summary>
+        /// <param name="hand">The hand exiting hover.</param>
+        protected virtual void OnHandHoverExitPO(ContactHand hand)
+        {
+            // Invoke event for hand hover exit
+            OnHandHoverExit?.Invoke(hand);
+        }
+
+        /// <summary>
+        /// Handles actions when a hand hovers over the pressable object.
+        /// </summary>
+        /// <param name="hand">The hand hovering.</param>
+        protected virtual void OnHandHoverPO(ContactHand hand)
+        {
+            // Invoke event for hand hover
+            OnHandHover?.Invoke(hand);
+        }
+
+        protected virtual void OnCollisionPO(Collision collision)
+        {
+            if (!collision.transform.root.GetComponent<PhysicalHandsManager>())
```

Comment:
If someone moved the PhysicalHandsManager to be a child of a "Scene assets" root, this would fail

We should check for whether it is a PhysicalHand collider a different way (perhaps layer?)